### PR TITLE
Add typeOp to MutableVisitor

### LINKED ISSUE #23: Mutable visitor doesn't support TypeOp
Code:
```kotlin
    val code = """
    val x = "" as String

""".trimIndent()
// Call the parser with the code
    val file = Parser.parseFile(code)

    MutableVisitor.preVisit(file) {v, _ -> v}
```

Results in:

```
Exception in thread "main" java.lang.IllegalStateException: Unrecognized node: TypeOp(lhs=StringTmpl(elems=[], raw=false), oper=Oper(token=AS), rhs=Type(mods=[], ref=Simple(pieces=[Piece(name=String, typeParams=[])])))
	at kastree.ast.MutableVisitor$visit$$inlined$run$lambda$1.invoke(MutableVisitor.kt:312)
	at kastree.ast.MutableVisitor$visit$$inlined$run$lambda$1.invoke(MutableVisitor.kt:3)
	at kastree.ast.MutableVisitor$ChangedRef.sub(MutableVisitor.kt:332)
```

Example is contrived, but it happens a lot in the codebase I'm looking at.

### PR DESCRIPTION
**Pull request by impatient**
Update tests to pass AST through identity transformed mutable visitor.

Resolves: https://github.com/cretz/kastree/issues/23

### CODE REVIEW

File: ast-psi/src/test/kotlin/kastree/ast/WriterTest.kt [hunk 1]
```diff
@@ -29,6 +29,23 @@ class WriterTest {
     }
 
     fun assertParseAndWriteExact(code: String) {
-        assertEquals(code.trim(), Writer.write(Parser.parseFile(code)).trim())
+
```
Comment (impatient) [inline ast-psi/src/test/kotlin/kastree/ast/WriterTest.kt:37]: Added the check here, so tests didn't have to be duplicated.  Had mutable visitor tests as a separate file, but made it too easy for tests to diverge.

File: ast/ast-common/src/main/kotlin/kastree/ast/MutableVisitor.kt [hunk 2]
```diff
@@ -270,6 +270,12 @@ open class MutableVisitor {
                         anns = visitChildren(anns, newCh),
                         expr = visitChildren(expr, newCh)
                     )
+                    is Node.Expr.TypeOp -> copy (
+                        lhs = visitChildren(lhs, newCh),
+                        oper = visitChildren(oper, newCh),
+                        rhs = visitChildren(rhs, newCh)
+                    )
+                    is Node.Expr.TypeOp.Oper -> this
```
Comment (impatient) [inline ast/ast-common/src/main/kotlin/kastree/ast/MutableVisitor.kt:278]: I tried a couple visitChildren variants here, but couldn't get it to work.  

I would assume that converting `as` to `as?` could be a valid case and this doesn't necessarily block that.

**Comment by RRua** [comment]
please merge this pull request

**Comment by cretz** [comment]
I will merge, but can't publish or tag soon (I don't use this lib in my day job as much)

**Pull request merged**